demonstrations/tutorial_pulse_programming101.py

```diff
@@ -0,0 +1,383 @@
+# sense to force the solver to evaluate the points of the evolution.
+# The error is still guaranteed to stay within the tolerance by using adaptive steps in between
+# the fixed ones we provide.
+ts = jnp.linspace(0., duration, t_bins)
+
+dev = qml.device("default.qubit", wires=range(n_wires))
+
+@qml.qnode(dev, interface="jax")
+def qnode(theta, t=ts):
+    qml.BasisState(list(data.tapered_hf_state), wires=H_obj.wires)
+    qml.evolve(H_pulse)(params=(*theta, *theta), t=t)
+    return qml.expval(H_obj)
+
+##############################################################################
+# We now have all the ingredients to run our ctrl-VQE program. We use the ``adabelief`` implementation in ``optax``, a package for optimizations in ``jax``.
+# The success of the optimization is sensitive to the initial values of the parameters.
+# We showcase a good random seed here. In reality, this optimization easily gets stuck in local minima
+# such that we would have to repeat the experiment multiple times with different random initializations. 
+# On the other hand, the Hartree-Fock state is usually a good starting point, i.e. choosing all parameters to zero.
+# However, this results in a near-zero gradient in our case. This is why we choose a trade-off by reducing
+# the initial amplitude of the random values.
+#
+# Further, the optimization is sensitive to the choice of optimizer and learning rate, which can be subject to
+# hyper parameter optimization. We here provide one of many possible choices leading to good results.
+
+key = jax.random.PRNGKey(999)
+theta = 0.01*jax.random.uniform(key, shape=jnp.array([n_wires, t_bins]))
+
+import optax 
+from datetime import datetime
+
+n_epochs = 100
+optimizer = optax.adabelief(learning_rate=1e-2)
+opt_state = optimizer.init(theta)
+
+value_and_grad = jax.jit(jax.value_and_grad(qnode))
+
+energy = np.zeros(n_epochs)
+cost = np.zeros(n_epochs)
+
+## Compile the evaluation and gradient function and report compilation time
+time0 = datetime.now()
+_ = value_and_grad(theta)
+time1 = datetime.now()
+print(f"grad and val compilation time: {time1 - time0}")
+
+## Optimization loop
+for n in range(n_epochs):
+    val, grad_circuit = value_and_grad(theta)
+    updates, opt_state = optimizer.update(grad_circuit, opt_state)
+    theta = optax.apply_updates(theta, updates)
+
+    energy[n] = qnode(theta)
+    cost[n] = val
+
+    if not n%10:
+        print(f"mean grad: {jnp.mean(grad_circuit)}")
+        print(f"{n+1} / {n_epochs}; energy: {energy[n]}; cost: {val}")
+
+##############################################################################
+# We see that we have converged well within chemical accuracy after half the number of epochs.
+
+fig, ax = plt.subplots(nrows=1, figsize=(5,3), sharex=True)
+
+y = np.array(energy) - E_exact
+ax.plot(y, ".:", label="$\\langle H_{{obj}}\\rangle - E_{{FCI}}$")
+ax.fill_between([0, len(y)-1], [1e-3]*2, 5e-4, alpha=0.2, label="chem acc.")
```

Comment:
I think this would look a tiny bit better:
```suggestion
ax.fill_between([0, len(y)], [1e-3]*2, 5e-4, alpha=0.2, label="chem acc.")
```